packages/git/src/browser/history/git-history-contribution.ts

```diff
@@ -43,6 +46,25 @@ export class GitHistoryContribution extends AbstractViewContribution<GitHistoryW
         });
     }
 
+    @postConstruct()
+    init() {
+        this.repositoryTracker.onDidChangeRepository(async repository => {
+            if (repository && repository.localUri) {
+                this.refreshWidget(repository.localUri);
+            }
+        });
+        this.repositoryTracker.onGitEvent(event => {
+            const { source, status, oldStatus } = event;
+            if (source && status && oldStatus) {
```

Comment:
Neither `source` not `status` are optional, they cannot be `undefined` if you are not hacking around. Why do you need to check them?